src/librustdoc/lib.rs

```diff
@@ -539,6 +541,9 @@ fn main_options(options: config::Options) -> MainResult {
         // if we ran coverage, bail early, we don't need to also generate docs at this point
         // (also we didn't load in any of the useful passes)
         return Ok(());
+    } else if run_check {
+        // Since we're in "check" mode, no need to generate anything beyond this point.
+        return Ok(());
```

Comment:
Are there any lints that run in `format`?

Reply:
There is one, moving it before. Great catch!

Reply:
Actually, it was run as expected. So no need to change anything. Added a test just in case to prevent regression though. ;)